Add support for policy API version 2

muntzerr opened the pull request.

davideschiera commented:
@muntzerr We had to do something similar for Sysdig Monitor dashboard API. In that case, we split the class implementation in v1 and v2 (where v1 would override methods to make the class backwards compatible). See v1 (https://github.com/draios/python-sdc-client/blob/master/sdcclient/_monitor_v1.py) and v2 (https://github.com/draios/python-sdc-client/blob/master/sdcclient/_monitor.py).

Would the same approach usable here? I'm just not sure about the possibility of adding many "duplicates" for each method as new API versions will become available, and the maintenance cost of doing so.

Let me know what you think. Thanks!

muntzerr commented:
I don't have a clear idea of the maintenance cost. I guess policy v1 can be deprecated at some point and then drop the `_secure_v1.py` or whatever.

I'll reimplement it as you suggest.

muntzerr commented:
Updated using a different class for each version of the API.

There are few incompatible changes:
* add|update_policy used to take just a json, now is a more explicit list of params. There is a new add|update_policy_json to import jsons directly.
* AFAIK get|set_policy_priorities are not meaningful anymore in V2. I removed them from the examples and SdSecureClient class.

muntzerr commented:
@davideschiera when you have the time to look at it tell me if this new version looks better or how can I improve it.

omer-sd commented:
add @tembleking and @davideschiera as reviewers so we can wrap this one up

davideschiera commented on sdcclient/_secure.py:
```diff
@@ -612,79 +628,81 @@ def list_policies(self):
+        return self._request_result(res)
 
-    def add_policy(self, policy_json):
+    def add_policy_json(self, policy_json):
```
What was the rationale behind adding the `_json` postfix? I'm asking because almost all functions in the libraries deal with JSON one way or another (parameters and outputs), so I wonder if the postfix is unnecessary.

muntzerr replied:
There are now two ways of adding a policy, one is passing the params separatelly (see [add_policy](https://github.com/draios/python-sdc-client/pull/99/files#diff-e6dd2abfa2a06413286a7994889acedcR675)) another is passing a json. This is why I added the `_json`, I need a way to differentiate the two methods. I'm happy to hear ideas on how to improve the naming.

davideschiera commented on sdcclient/_secure.py:
```diff
@@ -16,6 +16,17 @@ def __init__(self, token="", sdc_url='https://secure.sysdig.com', ssl_verify=Tru
+        self._policy_v2 = None
+
+    @property
+    def policy_v2(self):
```
Shouldn't this be implied by the fact that you instantiate `SdSecureClient` instead of `SdSecureClientV1`?

muntzerr replied:
The idea of this property is to be able to test if the provider support policy_v2 or not. With it is easier to write code that works with both versions of the policy API, something like:
```python
client = SdSecureclient()
if not client.policy_v2:
    client = SdSecureClientV1()
ok, res = client.list_policies()
...
```

davideschiera commented on test/test_secure_apis.sh:
```diff
@@ -112,25 +112,6 @@ if [[ $OUT = *"\"name\": \"Another Copy Of Write below binary dir\""* ]]; then
     exit 1
 fi
 
-WRITE_BELOW_BINARY_POS=`$SCRIPTDIR/../examples/list_policies.py $PYTHON_SDC_TEST_API_TOKEN | grep -b "\"name\": \"Write below binary dir" | awk -F: '{print $1}'`
```
Was there a specific reason to remove this section of tests?

muntzerr replied:
This test is testing the policy priorities, which AFAIK is deprecated in policy API V2.

omer-sd commented:
Since our SaaS is backward compatible with Alerts V1, @muntzerr - can you please make sure we are testing this with V1 as well to see nothing is broken?

muntzerr commented:
I just added tests for policy V1.
